test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -1618,6 +1618,122 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
+            $response.Headers.ContainsKey('Content-Range') | Should -BeFalse
+        }
+
+        It "Invoke-WebRequest -Resume overwrites existing file when remote server does not support resume" {
+            # Create a file larger than the download file
+            $largerFileSize = $referenceFileSize + 20
+            1..$largerFileSize | ForEach-Object { [Byte]$_ } | Set-Content -AsByteStream $outFile
+            $largerFileSize = Get-Item $outFile | Select-Object -ExpandProperty Length
+
+            $uri = Get-WebListenerUrl -Test 'Resume' -TestValue 'NoResume'
+            $response = Invoke-WebRequest -Uri $uri -OutFile $outFile -Resume -PassThru
+
+            $outFileHash = Get-FileHash -Algorithm SHA256 -Path $outFile
+            $outFileHash.Hash | Should -BeExactly $referenceFileHash.Hash
+            Get-Item $outFile | Select-Object -ExpandProperty Length | Should -Be $referenceFileSize
+            $response.Headers.'X-Has-Range'[0] | Should -BeExactly 'true'
+            $response.Headers.'X-Request-Range'[0] | Should -BeExactly "bytes=$largerFileSize-"
+            $response.StatusCode | Should -BeExactly 200
+            $response.Headers.ContainsKey('Content-Range') | Should -BeFalse
+        }
+
+        It "Invoke-WebRequest -Resume resumes downloading from <bytes> bytes" -TestCases @(
+            @{bytes = 4}
+            @{bytes = 8}
+            @{bytes = 12}
+            @{bytes = 16}
+        ) {
+            param($bytes, $statuscode)
+            # Simulate partial download
+            $uri = Get-WebListenerUrl -Test 'Resume' -TestValue "Bytes/$bytes"
+            $null = Invoke-WebRequest -uri $uri -OutFile $outFile
+            Get-Item $outFile | Select-Object -ExpandProperty Length | Should -Be $bytes
+
+            $uri = Get-WebListenerUrl -Test 'Resume'
+            $response = Invoke-WebRequest -uri $uri -OutFile $outFile -Resume -PassThru
+
+            $outFileHash = Get-FileHash -Algorithm SHA256 -Path $outFile
+            $outFileHash.Hash | Should -BeExactly $referenceFileHash.Hash
+            Get-Item $outFile | Select-Object -ExpandProperty Length | Should -Be $referenceFileSize
+            $response.Headers.'X-Has-Range'[0] | Should -BeExactly 'true'
+            $response.Headers.'X-Request-Range'[0] | Should -BeExactly "bytes=$bytes-"
+            $response.StatusCode | Should -BeExactly 206
+            $response.Headers.'Content-Range'[0] | Should -BeExactly "bytes $bytes-$($referenceFileSize-1)/$referenceFileSize"
+        }
+
+        It "Invoke-WebRequest -Resume assumes the file was successfully completed when the local and remote file are the same size." {
+            # Download the entire file
+            $uri = Get-WebListenerUrl -Test 'Resume' -TestValue 'NoResume'
+            $null = Invoke-WebRequest -uri $uri -OutFile $outFile
+            $fileSize = Get-Item $outFile | Select-Object -ExpandProperty Length
+
+            $uri = Get-WebListenerUrl -Test 'Resume'
```

Comment:
Maybe move to BeforeAll - we use this in almost every test.

Reply:
fixed